consensus/src/liveness/proposal_generator.rs

```diff
@@ -89,6 +87,54 @@ impl ChainHealthBackoffConfig {
     }
 }
 
+#[derive(Clone)]
+pub struct PipelineBackpressureConfig {
+    backoffs: BTreeMap<Round, PipelineBackpressureValues>,
+}
+
+impl PipelineBackpressureConfig {
+    pub fn new(backoffs: Vec<PipelineBackpressureValues>) -> Self {
+        let original_len = backoffs.len();
+        let backoffs = backoffs
+            .into_iter()
+            .map(|v| (v.back_pressure_pipeline_latency_limit_ms, v))
+            .collect::<BTreeMap<_, _>>();
+        assert_eq!(original_len, backoffs.len());
+        Self { backoffs }
+    }
+
+    #[allow(dead_code)]
```

Comment:
what you want is probably #[cfg(test)]

Reply:
leaving this for separate PR, it recursively needs to be changed, and then requires bunch of imports to have #[cfg(test)] as well, etc